packages/jest-circus/src/eventHandler.ts

```diff
@@ -113,7 +113,14 @@ const eventHandler: Circus.EventHandler = (
       }
       const parent = currentDescribeBlock;
 
-      currentDescribeBlock.hooks.push({asyncError, fn, parent, timeout, type});
+      currentDescribeBlock.hooks.push({
+        asyncError,
+        fn,
+        parent,
+        seenDone: false,
```

Comment:
I'm guessing it has something to do with this - this is only set when the hook is _defined_, but when it runs multiple times (like in a `beforeEach`) it is always `true` - we need to reset this when we start to _run_ a hook or test, not when it's defined.

Not sure why that would make the test hang rather than just fail, tho...

Reply:
I believe we'll need to have `seenDone` in state to keep track of how many times it was called if I'm not wrong. We'll have to go with resetting it when we start to run a hook or a test. Should I do it in `_callCircusHook` and `_callCircusTest`?

Reply:
> We might need to fail the test more explicitly than just throwing.

How can we do this? Looks like we'll need to, as multiple `done()` calls inside a `Promise` don't cause a test to fail as well.